### ISSUE TITLE
Upcoming release v0.2.0

### ISSUE THREAD

**Issue by uvchik** [oemof/oemof#382]
### Tests

 * [x] Storage init tests @uvchik
 * [x] tests for the outputlib (ask @gnn)
 * [x] Other init tests? @uvchik

### Code
 * [x] Relative vs. absolute imports #328 @uvchik
 * [x] Clean code (pylint, PEP8....)
 * [x] Use init.pys consistently? #328
 * [x] Integrating a license in all files. See: #262 and oemof/organisation#9 @uvchik
 * [x] Approach for saving/exporting/importing results @ckaldemeyer 
 * [x] Simulation with one time step? #306 @uvchik -> couldn't fix it -> postponed
 * [x] How to handle the open topics of #327 @simnh  @ckaldemeyer 

### Documentation:

 * [x] Getting started/Readme       @pkassing
    - [x] Examples section
 * [x] About oemof
    - [x] Update image @mloenneberga
 * [x] Installation and Setup
    - [x] Revise or remove "Additional Python packages" @c-moeller
    - [x] Revise "Run examples to check the installation" to new install check @mloenneberga
 * [x] Using oemof
    - [x] Revise network example @jnnr
    - [x] Add windpowerlib @birgits
    - [x] Revise feedinlib to new plans @birgits
    - [x] Plots moved to oemof-visio @jnnr
 * [x] Developing oemof
    - [x] See PR #319 -> "Install the developer version" @uvchik
    - [x] Revise Collaboration with Pull Requests @c-moeller
    - [x] Add "How to add a custom components". @c-moeller
 * [ ] Check Whats new file @uvchik
 * [x] oemof-network
    - [x] API changed (e.g. auto registration) please check. @c-moeller
 * [x] oemof-solph
    - [x] API changed, please check. @c-moeller
 * [x] oemof-outputlib
    - [x] API changed, please check. @jnnr
    - [x] Plots moved to oemof-visio @jnnr
 * [x] oemof-tools
    - [x] Revise according to PR #378
 
 * [x] fixed costs see #334 @SabineHaas

### Move examples to Examples-Repository:
 
 * [ ] Check, repair clean and document the examples and remove obsolete files:
   - [x] csv_reader @mloenneberga
   - [ ] sdewes_paper_2017: **Define input parameter that make sense!!!!**
   - [x] storage_invest @c-moeller 
   - [x] flexible_modelling 
   - [x] sector_coupling @birgits
   * [x] storage_investment @c-moeller @jnnr
   * [x] generic_chp @ckaldemeyer
   * [x] simple_dispatch @tbors
   * [x] variable_chp @s3pp
   * [x] plot example @uvchik 

 * [x] Create a README file with a short indroduction, a list of examples and space for short discriptions of each example (README should contain information on which example is supposed to work with which version) @birgits 
* [ ] Adapt GPL3-License file: https://github.com/oemof/organisation/issues/9

**Comment by uvchik** [comment]
@oemof/oemof-developer-group Please make sure that all changes should be done until December, 15th❗️

**Comment by uvchik** [comment]
@oemof/oemof-developer-group

There are still some ToDos without names. Your name means, that you are responsible. If you need help open an issue define Sub-Todos and ask for help.

We do need your help to release v0.2.0 before Christmas❗️

**Comment by uvchik** [comment]
Try to merge your new features before December 15th❗️

**Comment by c-moeller** [comment]
Which section is meant with oemof-network, oemof-solph, oemof-outputlib?
This one http://oemof.readthedocs.io/en/latest/using_oemof.html#oemof-network
or this one: http://oemof.readthedocs.io/en/latest/oemof_network.html# ?
as for the example of oemof-network

**Comment by uvchik** [comment]
There is one ToDo for the first one but this is already assigned to @jnnr (see `using_oemof.rst` with subsection oemof-network) and another Todo for the second one `oemof_network.rst`. I guess some changes will be the same in both sections.

You will find the same structure for all packages (network, solph, outputlib, tools).

And feel always free to add additional information you find interesting.

**Comment by uvchik** [comment]
In [oemof_examples](../../oemof_examples/issues/2#issuecomment-350696649) I introduced a proposal for an example header.

**Comment by uvchik** [comment]
@oemof/oemof-developer-group 

Please merge your stuff soon. Do not wait until the last minute otherwise we may have a last-minute-chaos with a bunch of merge conflicts.

You can rebase the new dev-branch commits to be aware of possible conflicts early.

**Comment by mloenneberga** [comment]
Considering your suggestion, at least for me raises the conflict, that possible API changes, like removing fixed_costs are being discussed a week before the release. This would effect basically all examples. Why not wait for 0.3? Is this really that urgent? (see #407 )

**Comment by simnh** [comment]
I can update the examples easily, more work are the lp-file test and the checks if everything works fine etc..

There is still the option to postpone the release (not that I want it). But I think we should rather use the current drive to get some things done, otherwise this won't happen a long time..

**Comment by ckaldemeyer** [comment]
> @oemof/oemof-developer-group
> 
> Please merge your stuff soon. Do not wait until the last minute otherwise we may have a last-minute-chaos with a bunch of merge conflicts.
> 
> You can rebase the new dev-branch commits to be aware of possible conflicts early.

I just talked to @uvchik 

> I can update the examples easily, more work are the lp-file test and the checks if everything works fine etc..
> 
> There is still the option to postpone the release (not that I want it). But I think we should rather use the current drive to get some things done, otherwise this won't happen a long time..

Should we prolong the deadline until the end of the year to not stop the current drive?

One idea would be to set the 31th as end day for all open PRs/changes. If something hasn't been finished until then, it will be omitted.

I would support this as I have some things I would like to bring in..

**Comment by uvchik** [comment]
I could live with that but I would encourage everybody to merge soon.

@oemof/oemof-developer-group 

❗️ As we just checked and adapted the documentation **every new PR** should make sure that it does **not contradict** anything written in the **actual documentation**.

Who will be available in the first week of January, to help with the release and some last minute todos?

**Comment by birgits** [comment]
> Who would be available in the first week of January, to help with the release and some last minute todos?

:hand:

**Comment by uvchik** [comment]
@c-moeller As you are assigned to the oemof-solph part of the documentation. As we do have the new constraint module we could add a constraint section in this part.

**Comment by c-moeller** [comment]
I created a pull request: #414 
What is the new constraint module? I could help with the documentation, but so far I don't know about the changes, so I need some help or anyone who knows more should add this.

**Comment by uvchik** [comment]
> What is the new constraint module? I could help with the documentation, but so far I don't know about the changes, so I need some help or anyone who knows more should add this.

A collection of additional constraints, see: https://github.com/oemof/oemof/blob/dev/oemof/solph/constraints.py

**Comment by uvchik** [comment]
@oemof/oemof-developer-group 

The deadline for the release is Sunday night (7th January, 2018)

 * Please make sure your issues/PRs are closed, merged or postponed
 * Please check for open reviews.
 * There are still some open todo's ([see above](https://github.com/oemof/oemof/issues/382#issue-278941540))
 * Please check your examples in the oemof-examples repository. We will merge the dev into the master next week
 * Assign yourself to the oemof groups (see oemof/organisation#23)
 * Ask for help if you have any problems!

**Comment by uvchik** [comment]
Here are some todo's left:

@mloenneberga : Please adept the image as mentioned above.

@ckaldemeyer : Please review PR #417, #409 and #334 and finish or postpone PR #358. Will you add your approach to save/export the results. If not, remove it from the list above and open an issue for e.g. v0.2.1. Please revise the generic_chp example and check the checkbox if you have finished.

@SabineHaas Please fix the merge conflict in PR #334.

@gnn What about #185, #187 and #336?

@tbors Please revise the simple_dispatch example and check the checkbox if you have finished.

**Comment by ckaldemeyer** [comment]
> Will you add your approach to save/export the results. If not, remove it from the list above and open an issue for e.g. v0.2.1.

No, I won't make it for the release.

**Comment by ckaldemeyer** [comment]
I have updated and fixed the generic chp examples!

**Comment by uvchik** [comment]
The release is coming closer. Most issues and PRs are closed or postponed.

Some last minute ToDos are left:

* This is the documentation that will be published with the release at the end of the week. Please check if all links/references work as expected (especially in parts that you revised)
http://oemof.readthedocs.io/en/release-v0_2_0/

* We will have some API changes with this release. Could anybody help to collect all API-changes. See the example below:

@oemof/oemof-assistants / @oemof/oemof-developer-group Could you help?

The most important thing is a list of all changes but sometimes a code example might be helpful.

## API-changes

### 1. Something changed
```diff
import oemof
- old code
+ new code
```
See the [documentation](http://oemof.readthedocs.io/en/latest/oemof_solph.html#oemof-solph-components-transformer-label) for more details.

### 2. Blub bla
....

**Comment by uvchik** [comment]
The release is closed. 
From now only bug-fixes and documentation improvements are allowed directly on the release branch -> `release/v0_2_0`.

**Comment by uvchik** [comment]
@oemof/oemof-developer-group Please add yourself to the whatsnew file if you have contributed between 0.1.4 and 0.2.0!

**Comment by uvchik** [comment]
@ckaldemeyer and myself have adapted the whatsnew file. Any additions?

http://oemof.readthedocs.io/en/release-v0_2_0/whats_new.html#v0-2-0-january-12-2018

Release will be on Friday. Last minute fixes (typos, grammar) still welcome.

**Comment by uvchik** [comment]
@oemof/oemof-developer-group 

From now on you can merge into the `dev` branch for v0.2.1

**Comment by uvchik** [comment]
🎉 DONE❗️

Thanks to @ckaldemeyer for his help with all the release stuff

Thanks to @birgits for some last changes and proofreading

Thanks to al contributors for your contribution (even though most of you do not [added their names](http://oemof.readthedocs.io/en/v0.2.0/whats_new.html#other-changes)).

Please help to promote the new release -> oemof/organisation#25

After the release is before the release so go ahead to add new features...

**Comment by ckaldemeyer** [comment]
> :tada: DONE:exclamation:
> 
> Thanks to @ckaldemeyer for his help with all the release stuff
> 
> Thanks to @birgits for some last changes and proofreading
> 
> Thanks to al contributors for your contribution (even though most of you do not added their names).
> 
> Please help to promote the new release -> oemof/organisation#25
> 
> After the release is before the release so go ahead to add new features...

Thanks a lot to you @uvchik for all your effort!! Have a nice weekend and enjoy your (hotfixing free) birthday :smile:
